src/Sylius/Bundle/AdminBundle/Action/Account/RenderResetPasswordPageAction.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Action\Account;
+
+use Sylius\Bundle\AdminBundle\Form\Type\ResetPasswordType;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\RouterInterface;
+use Twig\Environment;
+
+final class RenderResetPasswordPageAction
+{
+    public function __construct(
+        private UserRepositoryInterface $userRepository,
+        private FormFactoryInterface $formFactory,
+        private FlashBagInterface $flashBag,
+        private RouterInterface $router,
+        private Environment $twig,
+        private string $tokenTtl,
+    ) {
+    }
+
+    public function __invoke(Request $request, string $token): Response
+    {
+        /** @var AdminUserInterface|null $admin */
+        $admin = $this->userRepository->findOneBy(['passwordResetToken' => $token]);
+        if (null === $admin) {
+            throw new NotFoundHttpException('Token not found');
+        }
+
+        $lifetime = new \DateInterval($this->tokenTtl);
+
+        if (!$admin->isPasswordRequestNonExpired($lifetime)) {
+            $this->flashBag->add('error', 'sylius.admin.password_reset.token_expired');
+
+            $attributes = $request->attributes->get('_sylius');
+            $redirect = $attributes['redirect'] ?? 'sylius_admin_login';
+
+            if (is_array($redirect)) {
```

Comment:
As far as I see it would just go further if the redirect is not an array? I feel something is missing here 😄

Reply:
Must have yeeted it accidentally during rebase 🥏 
Applied in #14153